tests/Common/ComponentsTest.php

```diff
@@ -464,31 +517,47 @@ public function testComponentConfigRestrictionsForReadOnlyUser()
         $this->assertSame($components, $componentsForAdmin->listComponents());
     }
 
-    public function testConfigurationNameShouldBeRequired()
+    /**
+     * @dataProvider componentsProvider
+     */
+    public function testConfigurationNameShouldBeRequired(callable $getComponents)
     {
+        /** @var \Keboola\StorageApi\Components $components */
+        $components = $getComponents($this->_client);
         try {
-            $this->_client->apiPost('components/wr-db/configs', []);
+            $components->getClient()->apiPost('components/wr-db/configs', []);
```

Comment:
tohle bys nemusel řešit kdybys z dataprovideru dostal $client

Reply:
plus volání klienta přímo je edge case, který by klidně dataprovider nemusel podporovat IMHO

Reply:
Jo, toho jsem si nevsim, upraveno, ze vraci Client.